docs/protocols/filter_gadget.tex

```diff
@@ -0,0 +1,57 @@
+\documentclass[11pt]{article}
+\usepackage[T1]{fontenc}
+\usepackage[utf8]{inputenc}
+\usepackage{lmodern}
+\usepackage[margin=1in]{geometry}
+\usepackage{graphicx}
+\usepackage{amsmath,amssymb}
+\usepackage{booktabs}
+\usepackage{hyperref}
+\usepackage{microtype}
+\usepackage{todonotes}
+\hypersetup{
+  colorlinks=true,
+  linkcolor=blue,
+  citecolor=blue,
+  urlcolor=blue
+}
+\setlength{\parindent}{0pt}
+\setlength{\parskip}{6pt}
+
+\title{Filter Gadget}
+\author{MakeInfinite Inc}
+\date{June 2025}
+
+\begin{document}
+\maketitle
+
+\section{Definition}
+\noindent Let $(C_j)$ be a collection of $a$ columns each of length $n$. Let $S=(s_i)$ be a column of only $1$s and $0$s of length $n$. Define the function $f$ so that $f(C_j, S)$ is the column of all $c_{j,i}$ such that $s_i=1$. A collection of $a$ columns $(R_j)$, each of length $m$, is $(f(C_j, S))$ if and only if the following constraints hold.
```

Comment:
Hmm it is kind of weird to refer to the num of columns in a table as a. Maybe something like k will be better?